File: src/main/java/com/rdvdev2/timetravelmod/impl/common/timemachine/exception/TimeMachineExecutionException.java
package com.rdvdev2.timetravelmod.impl.common.timemachine.exception;

import com.rdvdev2.timetravelmod.impl.common.timemachine.TimeMachineExecutor;

public class TimeMachineExecutionException extends Exception {

    private final TimeMachineExecutor.TimeMachineError error;

    public TimeMachineExecutionException(TimeMachineExecutor.TimeMachineError error) {
        this.error = error;
    }

    public TimeMachineExecutor.TimeMachineError getError() {
        return error;
    }
}

File: src/main/kotlin/com/rdvdev2/timetravelmod/impl/common/timemachine/exception/TimeMachineExecutionException.kt
package com.rdvdev2.timetravelmod.impl.common.timemachine.exception

import com.rdvdev2.timetravelmod.impl.common.timemachine.TimeMachineExecutor.TimeMachineError

class TimeMachineExecutionException(val error: TimeMachineError) : RuntimeException()